Added custom training container examples, based on the Amazon SageMaker training toolkit. (#1163)
awslabs/amazon-sagemaker-examples: sagemaker-custom-training-containers/README.md

# Amazon SageMaker Custom Training containers

> This folder contains skelethon implementations of Amazon SageMaker-compatible training containers.

Reviewer: ```suggestion
This folder contains skeleton implementations of Amazon SageMaker-compatible training containers.
```

> The purpose of these examples is explaining how to build a custom container for training, with a relevant focus on the Amazon SageMaker Training Toolkit, a toolkit that facilitates the development of training containers for SageMaker, and enables dynamic loading of user scripts from Amazon S3, thus separating the execution environment (Docker container) from the script being executed. For additional info please see: [https://github.com/aws/sagemaker-training-toolkit](https://github.com/aws/sagemaker-training-toolkit).

Reviewer: ```suggestion
The purpose of these examples is to explain how to build a custom container for training using the Amazon SageMaker Training Toolkit. This toolkit facilitates the development of SageMaker-compatible training containers and enables dynamic loading of user scripts from Amazon S3, thus separating the execution environment (Docker container) from the script being executed. For additional info, please see: [https://github.com/aws/sagemaker-training-toolkit](https://github.com/aws/sagemaker-training-toolkit).
```

> By purpose, no specific ML science is applied in this context, and code is simulating training dummy models.

Reviewer: ```suggestion
These examples deliberately do not use any specific ML framework or algorithm. The custom training container functionality is demonstrated by training dummy models.
```
